uint8 senderID
uint8 recID
uint32 flags
string programName
float64[6] positions
uint8 val